bug_actiongroup.php

```diff
@@ -206,19 +208,27 @@
 		case 'UP_STATUS':
 			$f_status = gpc_get_int( 'status' );
 			if( access_has_bug_level( access_get_status_threshold( $f_status, $t_bug->project_id ), $t_bug_id ) ) {
-				if( true == bug_check_workflow( $t_status, $f_status ) ) {
+				if( true == bug_check_workflow( $t_status, $f_status ) ||
+					access_compare_level( current_user_get_access_level(), config_get_access( 'status_enum_workflow' ) ) ) {
 					# @todo we need to issue a helper_call_custom_function( 'issue_update_validate', array( $t_bug_id, $t_bug_data, $f_bugnote_text ) );
-					bug_set_field( $t_bug_id, 'status', $f_status );
-
-					# Add bugnote if supplied
-					if( !is_blank( $f_bug_notetext ) ) {
-						bugnote_add( $t_bug_id, $f_bug_notetext, null, $f_bug_noteprivate );
-						# No need to call email_generic(), bugnote_add() does it
+					if( $t_status !== $f_status ) {
+						bug_set_field( $t_bug_id, 'status', $f_status );
+						if( is_blank( $f_bug_notetext ) ) {
+							$t_new_status_label = MantisEnum::getLabel( config_get( 'status_enum_string' ), $f_status );
+							$t_new_status_label = str_replace( ' ', '_', $t_new_status_label );
+							email_bug_status_changed( $t_bug_id, $t_new_status_label );
+						} else {
+							bugnote_add( $t_bug_id, $f_bug_notetext, null, $f_bug_noteprivate );
+							# No need to call email_generic(), bugnote_add() does it
+						}
+						helper_call_custom_function( 'issue_update_notify', array( $t_bug_id ) );
 					} else {
-						email_bug_updated( $t_bug_id );
+					 	if( !is_blank( $f_bug_notetext ) ) {
+							bugnote_add( $t_bug_id, $f_bug_notetext, null, $f_bug_noteprivate );
+							helper_call_custom_function( 'issue_update_notify', array( $t_bug_id ) );
+						}
+						$t_failed_ids[$t_bug_id] = lang_get( 'bug_actiongroup_same_status' );
```

Comment:
Why do we consider the case where an issue is set to the same status as a failure?  The goal is to have a group in state X and if some are already there, then that is already a success.  I would remove this + the associated string.